ipu-plugin/pkg/ipuplugin/lifecycleservice.go

```diff
@@ -685,7 +685,15 @@ func countAPFDevices() int {
 	return len(pfList)
 }
 
-func checkIfMACIsSet() (bool, string) {
+/*
+	IMC reboot needed for following cases:
+
+1. First time provisioning of IPU system(where MAC gets set in node policy)
+2. Upgrade-(like P4 package update).
+3. (TODO) Upgrade- other change in node policy thro load_custom_pkg.sh.
+For now, supporting case1, 2 above.
+*/
+func checkIfIMCRebootIsRequired() (bool, string) {
```

Comment:
Can we please add the meaning of the ret vals here in the comments? The previous meaning I think was 
bool = Whether Mac is set
string = Error string
The variable names in the place where the function is being called are confusing too. They didn't match actual expectations earlier either